lib/saputils.pm

```diff
@@ -31,63 +32,146 @@ SAP Applications.
-            sync_state => 'SOK',
-            vhost => 'vmhana02',
-        },
-    }
+          'Host' => {
+                      'vmhana01' => {
+                                           'clone_state' => 'DEMOTED',
+                                           'score' => '100',
+                                           'srah' => '-',
+                                           'version' => '2.00.077.00.1710325774',
+                                           'vhost' => 'vmhana01',
+                                           'site' => 'site_a'
+                                         },
+                      'vmhana02' => {
+                                           'clone_state' => 'PROMOTED',
+                                           'score' => '150',
+                                           'vhost' => 'vmhana02',
+                                           'srah' => '-',
+                                           'version' => '2.00.077.00.1710325774',
+                                           'site' => 'site_b'
+                                         }
+                    },
+          'Site' => {
+                      'site_a' => {
+                                     'srMode' => 'sync',
+                                     'srPoll' => 'SOK',
+                                     'mns' => 'vmhana01'
+                                   },
+                      'site_b' => {
+                                     'srPoll' => 'PRIM',
+                                     'mns' => 'vmhana02',
+                                     'srMode' => 'sync'
+                                   }
+                    }
+        };
+
+
 
 =over 1
 
-=item B<input> - stdout of 'SAPHanaSR-showAttr --format=script'
+=item B<input_format> - format of the 'SAPHanaSR-showAttr --format='
+
+=item B<input> - stdout of 'SAPHanaSR-showAttr --format=<input_format>'
 
 =back
 =cut
 
 
 sub calculate_hana_topology {
     my (%args) = @_;
-    croak("Argument <input> missing") unless $args{input};
+    croak('calculate_hana_topology [ERROR] Argument <input> missing') unless $args{input};
+    my $input_format = $args{input_format} || 'script';
+    croak("calculate_hana_topology [ERROR] Argument <input_format: $input_format > is not known") unless ($input_format eq 'script' or $input_format eq 'json');
+
     my %topology;
-    my @all_parameters = map { if (/^Hosts/) { s,Hosts/,,; s,",,g; $_ } else { () } } split("\n", $args{input});
-    my @all_hosts = uniq map { (split("/", $_))[0] } @all_parameters;
-
-    for my $host (@all_hosts) {
-        # Only takes parameter and value for lines about one specific host at time
-        my %host_parameters = map {
-            my ($node, $parameter, $value) = split(/[\/=]/, $_);
-            if ($host eq $node) { ($parameter, $value) } else { () }
-        } @all_parameters;
-        $topology{$host} = \%host_parameters;
-    }
+    my $topology_json;
+    my %script_topology;
```

Comment:
What is `script_topology`? What is its role? Please mind to eventually add a small comment in the code about it?

It seems to me some sort of intermediate structure to populate `%topology`, that sometime is used to get keys for `topology`

```
$topology{'Site'}{$script_topology{$host}->{'site'}}{'lss'} = ($script_topology{$host}->{'node_state'} eq 'online' or $script_topology{$host}->{'node_state'} =~ /[1-9]+/) ? '4' : '1';
```

and other times to get values
```
 $topology{'Host'}{$host}{'vhost'} = $script_topology{$host}->{'vhost'};
```

How it works?

Reply:
[SOLVED] comment added in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21264/files#diff-4a40a74cf6048efd08f8d55e2613d752fb683e6dc56fc331da7e8f23ac0af1e9R138-R140